### ISSUE TITLE
Error on Travis CI against Oracle when creating #temp_dates.

### ISSUE THREAD

**Issue by chrisknoll** [OHDSI/Achilles#191]
### Expected behavior
Tests should execute without error on oracle.

### Actual behavior
Error reported:
```sql
Error:
execute JDBC update query failed in dbSendUpdate (ORA-00909: invalid number of arguments
)

SQL:
CREATE TABLE 
  OHDSI.xp5wfdi2temp_dates
  AS
SELECT
   DISTINCT 
  EXTRACT(YEAR FROM observation_period_start_date) AS obs_year,
  TO_DATE(CONCAT(TO_CHAR(EXTRACT(YEAR FROM observation_period_start_date) ), '01', '01') , 'yyyymmdd') AS obs_year_start,
  TO_DATE(CONCAT(TO_CHAR(EXTRACT(YEAR FROM observation_period_start_date) ), '12', '31') , 'yyyymmdd') AS obs_year_end
FROM
  CDMV5.observation_period
```

### Steps to reproduce behavior
Error appears during travis CI build operation.

**Comment by chrisknoll** [comment]
According to https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions026.htm, CONCAT() on oracle does not work the same way as CONCAT() on mssql or postgresql.  On mssql and postgresql, these functions can take a variable number of parameters.  On oracle, it concats exactly 2 strings.

It looks like sql render doesn't have a translation rule for CONCAT() that will map a mssql style CONCAT to a  oracle style concat.  That's why we've used '+' as a concatenation prior to Impala (which doesn't have a string concatinate operator, but does have a function.

So the fix here is to revert the CONCAT change from impala and find an appropriate workaround for impala concatination.  One possible generic solution is to use CONCAT() mssql form (which is supported equally well in postgresql and mssql) and figure out a way to have CONCAT(a,b,c,d) in oracle to become CONCAT(a,CONCAT(b,CONCAT(c,d)), or for impala have a+b+c+d turn into concat(a,b,c,d).

But the priority here is to get the oracle build to pass the tests, and rework the impala implementation after.

**Comment by t-abdul-basser** [comment]
@chrisknoll Good catch. That certainly seems to be the root cause. Agreed concerning proposed course of action. To  my mind, this strengthens the case for your proposed course of action in comments in #187.
